### ISSUE TITLE
Character encodings in import 

### ISSUE THREAD

**Issue by timbl** [NoelDeMartin/umai#17]
https://umai.noeldemartin.com/recipes/zuumlrcher-geschnetzeltes-cut-meat-zuumlrich-style

`Z&uuml;rcher Geschnetzeltes (Cut Meat Z&uuml;rich Style)`

due to some problem with character encodings when it was imported from

https://www.internationalcuisine.com/zurich-style-veal-in-cream-sauce/

**Comment by NoelDeMartin** [comment]
Hey, thanks for opening an issue!

Is that the website you used to import the recipe? Because it doesn't seem to have the same title, when I import it it's called "Zurich Style Veal in Cream Sauce (Zurcher Geschnetzeltes)", without any ü.

In any case, I have tried with other websites using that character and it seems to be working fine, could it be that the website you used already had that problem in their JSON-LD or HTML? Maybe I can take a look anyways, but the app should work with these characters because I already fixed that in #8.

**Comment by timbl** [comment]
Interesting.  If I try it again I see a "import from the web" panel with it wrong: 
<img width="908" alt="image" src="https://github.com/user-attachments/assets/f5687618-c295-41a8-bf78-cba3a711099d">

**Comment by timbl** [comment]
The website has it wrong in the title, 
<img width="216" alt="image" src="https://github.com/user-attachments/assets/74aa935b-59ce-49da-bba6-ca49175625df">
though not in the page. 
<img width="896" alt="image" src="https://github.com/user-attachments/assets/3781e31a-cdb3-4d19-9ac5-16ec2285cb40">

**Comment by timbl** [comment]
and even in the html
```
<title>Z&amp;uuml;rcher Geschnetzeltes (
```

**Comment by timbl** [comment]
I guess this is food.com's problem.
Sorrry to set you on a wild goose chase

**Comment by NoelDeMartin** [comment]
Thanks for pointing to `food.com`, I could find the recipe and indeed it seems like it's a problem on their side :(.

In this case, Umai is getting the info from an `application/ld+json` script that is embedded in the page, and it has the wrong format:

```json
 {"@context":"http://schema.org","@type":"Recipe","mainEntityOfPage":"true","name":"Z&uuml;rcher Geschnetzeltes - Zurich Veal Recipe","author":"PanNan","cookTime":"PT30M","prepTime":"PT10M","totalTime":"PT40M","datePublished":"2011-05-14T20:45Z","description":"Make and share this Z&uuml;rcher Geschnetzeltes - Zurich Veal Recipe recipe from Food.com.","image":"https://img.sndimg.com/food/image/upload/q_92,fl_progressive,w_1200,c_scale/v1/img/recipes/45/61/28/picOHkry1.jpg","recipeCategory":"Veal","keywords":"Meat,Swiss,European,< 60 Mins","recipeIngredient":["2   lbs    veal cutlets, sliced into 3/4 in strips ","3   tablespoons    all-purpose flour","2   ounces    unsalted butter","2   tablespoons    olive oil","1       onion, very finely sliced ","2       garlic cloves, finely sliced ","15       fresh sage leaves, chopped ","7   ounces    cremini mushrooms, thickly sliced ","1   cup    dry white wine"," sea salt & freshly ground black pepper","1 1/4  cups    heavy cream"],"aggregateRating":{"@type":"AggregateRating","ratingValue":"5.0","reviewCount":"7"},"nutrition":{"@type":"NutritionInformation","calories":"558.2","fatContent":"40.8","saturatedFatContent":"21.1","cholesterolContent":"212.2","sodiumContent":"148.9","carbohydrateContent":"8.7","fiberContent":"0.6","sugarContent":"1.8","proteinContent":"31.9"},"recipeInstructions":[{"@type":"HowToStep","text":"Sprinkle the flour over the meat and coat well. Heat half of both the butter and oil in a frying pan. As soon as the butter foams, add the meat and brown it very quickly on all sides. Set aside on a plate."},{"@type":"HowToStep","text":"Add the rest of the butter and oil to the pan and saut&eacute; the onion, garlic and sage for a few minutes, then add the mushrooms. Stir well and when the mushrooms start to soften, continue cooking for about 5 minutes longer."},{"@type":"HowToStep","text":"Add the wine to deglaze the bottom of the pan, using a wooden spoon to scrape away any crust that has formed. Add the veal and cook, stirring, until the wine has evaporated. Turn the heat down to low and cook for another 10 minutes."},{"@type":"HowToStep","text":"Season with salt and pepper, and at the last moment, add the cream, let it bubble for 2 minutes, then remove from the heat and serve."}],"recipeYield":"6 serving(s)","review":[{"@type":"Review","description":"Wow!  This was delicious!  I'd never had veal before, but it was on sale at my grocery store, and I couldn't pass it up.  It was a bit tough cutting up the veal, but it was worth the work.  I did make some substitutions for ingredients I didn't have:  a veal shoulder chop instead of cutlets, half & half for the cream (added a little extra butter to boost it up), oregano for the sage (I don't use sage often enough to have it on hand regularly), and chicken broth for the wine.  Other than those few things, I followed the recipe exactly.  I served it on cooked rotini, as I did not have time to make the rosti.  I plan to make this recipe again, though, and I will definitely make the rosti next time.<br/><br/>I think I could eat the sauce, onions, garlic, and mushrooms on just about anything.  As good as the veal was, that part of the dish was even better.  I'll probably have dreams of it tonight!","datePublished":"August 09, 2012","itemReviewed":{"@type":"Thing","name":"Z&uuml;rcher Geschnetzeltes - Zurich Veal Recipe"},"reviewRating":{"@type":"Rating","worstRating":"1","ratingValue":5,"bestRating":"5"},"author":"elise6185"}],"publisher":{"@type":"Organization","name":"Food.com","logo":{"@type":"ImageObject","url":"https://geniuskitchen.sndimg.com/fdc-new/img/FDC-Logo.png"},"url":"https://www.food.com"}}
```

I guess I could do something in the app to fix this, I'll take a look.

**Comment by NoelDeMartin** [comment]
Ok, I've fixed this in version [v0.1.7](https://github.com/NoelDeMartin/umai/blob/main/CHANGELOG.md#v017---2024-12-03).

Even though it was a problem on their part, I think it's fairly safe to decode html entities when they are present. If I come across some unintended behaviour, I may revert it. But for now, it should work :).
